src/Playwright.Tests/FrameHierarchyTests.cs

```diff
@@ -191,8 +191,11 @@ public async Task ShouldReportDifferentFrameInstanceWhenFrameReAttaches()
             }");
             Assert.True(frame1.IsDetached);
 
+            var frameEvent = new TaskCompletionSource<IFrame>();
+            Page.FrameNavigated += (_, frame) => frameEvent.TrySetResult(frame);
```

Comment:
Comparing the two changes in this file. How would a customer know when to use `WaitForNavigationAsync` and when they would need to implement that mannualy?